### PR TITLE
test: add simulations for VAI upgrade and post-upgrade behavior

### PR DESCRIPTION
**Pull request by kkirka**
This PR builds on top of https://github.com/VenusProtocol/venus-protocol/pull/192 and brings in utility functions for simulating VIPs, and a simulation for VAI stability fee pugrade

### CODE REVIEW

File: script/hardhat/fork/govUpdateTest.ts [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+import { expect } from "chai";
+import { parseUnits } from "ethers/lib/utils";
+import { ethers } from "hardhat";
+
+import { Comptroller, GovernorBravoDelegate, IAccessControlManager } from "../../../typechain";
+import { TimelockInterface } from "../../../typechain/contracts/Governance/GovernorAlpha2.sol";
+import { getCalldatas, setForkBlock } from "./vip-framework/utils";
+
+const ONE_HOUR = 3600; // 1 hour
+const SIX_HOURS = 6 * 3600; // 6 hours
```
Comment (coreyar) [inline script/hardhat/fork/govUpdateTest.ts:12]: nit: could multiply the constant, which will be nice in the event the number of seconds in an hour changes 😝 
```suggestion
const SIX_HOURS = 6 * ONE_HOUR; // 6 hours
```

File: script/hardhat/fork/govUpdateTest.ts [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+let timeLockFastTrack: TimelockInterface;
+let timeLockCritical: TimelockInterface;
+
+const initMainnetUser = async (user: string, balance: number) => {
+  await impersonateAccount(user);
+  await setBalance(user, balance);
+  return ethers.getSigner(user);
+};
+const governanceFixture = async (): Promise<void> => {
+  //Mandatory when creating forked tests. One must specify form which block the fork should be
```
Comment (coreyar) [inline script/hardhat/fork/govUpdateTest.ts:68]:
```suggestion
  // Mandatory when creating forked tests. One must specify form which block the fork should be
```

**Review by chechu — COMMENTED** [review]
Awesome job! Would it make sense to move the vip-framework to an isolated package (maybe in the future)? I was wondering if we could use it in the future for VIP's related to isolated pools

**Comment by kkirka** [comment]
> Awesome job! Would it make sense to move the vip-framework to an isolated package (maybe in the future)? I was wondering if we could use it in the future for VIP's related to isolated pools

I'd rather move the whole governance into its own package. Governance is something that is relevant not only to core, but also to isolated pools and oracle, and maybe other contracts we will have in future.

**Review by chechu — APPROVED** [review]

**Comment by chechu** [comment]
> I'd rather move the whole governance into its own package. Governance is something that is relevant not only to core, but also to isolated pools and oracle, and maybe other contracts we will have in future.
> 
> This would also allow us to version governance independently from core/IL/oracle.

https://jira.toolsfdg.net/browse/VEN-930

**Pull request merged**